src/lib/y2security/autoinst/lsm_config_reader.rb

```diff
@@ -18,55 +18,47 @@
 # find current contact information at www.suse.com.
 
 require "y2security/lsm/config"
-require "y2security/autoinst_profile"
+require "y2security/autoinst_profile/security_section"
 
 module Y2Security
   module Autoinst
     # This class is responsible of reading the Linux Security Module configuration declared in
     # the AutoYaST profile
     class LSMConfigReader
-      # @return [AutoinstProfile::LSMSection]
+      # @return [AutoinstProfile::SecuritySection]
       attr_reader :section
-      # @return [AutoinstProfile::SelinuxSection, AutoinstProfile::ApparmorSection, nil]
-      attr_reader :module_section
-
       # Constructor
       #
-      # @param section [AutoinstProfile::LSMSection]
+      # @param section [AutoinstProfile::SecuritySection]
       def initialize(section)
         @section = section
       end
 
       # Reads the Linux Security Module configuration defined in the profile modifying it
       # accordingly
       def read
-        return unless section
+        return unless section.lsm_select || section.selinux_mode
 
-        config.configurable = section.configurable
-        config.select(section.select) if section.select
-        configure_supported_modules
+        select_module
+        configure_selinux if selinux?
       end
 
     private
 
-      def configure_supported_modules
-        [:selinux, :apparmor, :none].each do |id|
-          lsm_module = config.public_send(id)
-          @module_section = section.public_send(id)
-          next unless module_section
+      def selinux?
+        return true if section.lsm_select == "selinux"
+        return true if !section.lsm_select && section.selinux_mode
```

Comment:
I expect rubocop will complain that you can write it simply as `!section.lsm_select && section.selinux_mode` without return, if and false

Reply:
and please explain why ( I expect backward compatibility? or reasonable default? )

Reply:
> I expect rubocop will complain that you can write it simply as `!section.lsm_select && section.selinux_mode` without return, if and false

Rubocop does not complain but you are right.

Reply:
> and please explain why ( I expect backward compatibility? or reasonable default? )

I would say that It is the same behavior than before #115 was introduced. If no LSM is selected explicitly then it will select SELinux if the **selinux_mode** is declared.